demo/src/lib.rs

```diff
@@ -9,13 +9,14 @@ use crate::plugins::ImagesPluginData;
 use egui::Context;
 use walkers::{HttpOptions, Map, MapMemory, Tiles, TilesManager};
 
-#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+#[derive(Debug, Clone, PartialEq, Eq, Hash)]
 pub enum Provider {
     OpenStreetMap,
     Geoportal,
     MapboxStreets,
     MapboxSatellite,
     LocalTiles,
+    Custom(String),
```

Comment:
I do appreciate adding new tile source, but I am still quite confused why is it so different than other ones here.

Reply:
The whole concept of this PR is:
- I want to use the `walkers` library
- I want to use my own tiles provider
   - option 1 do on the PR on this repository (may be slow, or not possible if provider if private)
   - option 2 do like this PR by creating your own tilesManager

Reply:
I am sorry, but I am still missing the intention here. Do you want to provide an example (this is what `demo` is for - an example) on how to create your own TileSource, or do you want to make it possible to do so?

Reply:
I added with provider to the lib and changed the pr